apps/core/middlewares/with-custom-urls.ts

```diff
@@ -32,16 +32,44 @@ const createRewriteUrl = (path: string, request: NextRequest) => {
   return url;
 };
 
+const isRouteCache = (
```

Comment:
Thoughts on using `zod` instead of these custom type-guards?

Reply:
my thoughts are that I will use `zod` instead of these custom type-guards